integration-tests/upgradeable-contracts/delegator/lib.rs

```diff
@@ -204,5 +228,57 @@ pub mod delegator {
 
             Ok(())
         }
+
+        #[ink_e2e::test]
+        async fn update_delegate<Client: E2EBackend>(
+            mut client: Client,
+        ) -> E2EResult<()> {
+            // given
+            let origin = client
+                .create_and_fund_account(&ink_e2e::alice(), 10_000_000_000_000)
+                .await;
+
+            let code_hash = client
+                .upload("delegatee", &origin)
+                .submit()
+                .await
+                .expect("upload `delegatee` failed")
+                .code_hash;
+
+            let code_hash2 = client
+                .upload("delegatee2", &origin)
+                .submit()
+                .await
+                .expect("upload `delegatee2` failed")
+                .code_hash;
+
+            let mut constructor = DelegatorRef::new(10, code_hash);
+            let contract = client
+                .instantiate("delegator", &origin, &mut constructor)
+                .submit()
+                .await
+                .expect("instantiate failed");
+            let mut call_builder = contract.call_builder::<Delegator>();
+
+            // when
+            let call_delegate = call_builder.update_delegate_to(code_hash2);
+            let result = client.call(&origin, &call_delegate).submit().await;
+            assert!(result.is_ok(), "update_delegate_to failed.");
+
+            // then
+
+            // remove the original delegatee code.
+            // should succeed because the delegate dependency has been removed.
+            let original_code_removed =
+                client.remove_code(&origin, code_hash).submit().await;
+            assert!(original_code_removed.is_ok());
+
+            // attempt to remove the new delegatee code.
+            // should fail because of the delegate dependency.
+            let new_code_removed = client.remove_code(&origin, code_hash2).submit().await;
```

Comment:
Need to test that the alternative account (i.e. bob) can not remove the dependency

Reply:
Or rather that they *can* remove the dependency?

Reply:
Ah, okay, didn't know that